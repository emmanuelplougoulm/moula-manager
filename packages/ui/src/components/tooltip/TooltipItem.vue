<template>
  <div
    class="tooltip-container"
    data-testid="tooltip"
    @mouseover="showTooltip"
    @touchstart="showTooltip"
    @mouseleave="hideTooltip"
    @touchend="hideTooltip"
  >
    <Info class="info-icon" />
    <div v-show="state.isTooltipVisible" class="tooltip">
      <div v-if="props.title" class="tooltip-title text-small-semi-bold">
        {{ props.title }}
      </div>
      <div class="tooltip-description text-small-regular">
        {{ props.description }}
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { reactive } from 'vue';
import Info from '../icons/Info.vue';

const state = reactive({
  isTooltipVisible: false
});

const showTooltip = () => {
  state.isTooltipVisible = true;
};

const hideTooltip = () => {
  state.isTooltipVisible = false;
};

const props = defineProps({
  text: {
    type: String,
    default: ''
  },
  title: {
    type: String,
    default: ''
  },
  description: {
    type: String,
    default: ''
  }
});
</script>

<style scoped>
.tooltip-container {
  z-index: 99;
}
.info-icon {
  width: 1.5rem;
  height: 1.5rem;
  color: var(--primary-30);
}

.tooltip-title {
  color: var(--neutral-10);
  line-height: 1.125rem;
}
.tooltip-description {
  color: var(--neutral-20);
  line-height: 1.125rem;
}

.tooltip {
  display: flex;
  flex-direction: column;
  gap: 0.25rem;
  padding: 0.75rem;
  background-color: #ffffff;
  box-shadow: var(--box-shadow);
  border-radius: 0.25rem;
  word-wrap: break-word;
  position: absolute;
  right: 0;
  left: 0;
  bottom: 2rem;
  z-index: 999;
}
.tooltip:after {
  content: '';
  position: absolute;
  bottom: -0.5rem;
  right: 0.75rem;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 1.25rem 1.25rem 0;
  border-color: transparent #ffffff transparent transparent;
}
</style>
