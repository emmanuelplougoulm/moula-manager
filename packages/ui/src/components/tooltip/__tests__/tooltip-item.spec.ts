import { mount } from '@vue/test-utils';
import TooltipItem from '../TooltipItem.vue';

const mouseOverEvent = 'mouseover';
const tooltipId = "[data-testid='tooltip']";

describe('TooltipItem tests', () => {
  it('should display tooltip with a title and a description when hovering the info icon', () => {
    const wrapper = mount(TooltipItem, {
      props: {
        title: 'Questo è un titolo',
        description: 'Questa è una descrizione'
      }
    });

    wrapper.find(tooltipId).trigger(mouseOverEvent);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not display tooltip since the object there is no title nor description passed as props', () => {
    const wrapper = mount(TooltipItem);
    wrapper.find(tooltipId).trigger(mouseOverEvent);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display a tooltip on @mouseover and hide it on @mouseleave', () => {
    const wrapper = mount(TooltipItem, {
      props: {
        title: 'Questo è un titolo',
        description: 'Questa è una descrizione'
      }
    });

    wrapper.find(tooltipId).trigger(mouseOverEvent);
    wrapper.find(tooltipId).trigger('mouseleave');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
