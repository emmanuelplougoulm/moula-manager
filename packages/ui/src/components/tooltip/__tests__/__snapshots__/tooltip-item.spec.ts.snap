// Vitest Snapshot v1

exports[`TooltipItem tests > should display a tooltip on @mouseover and hide it on @mouseleave 1`] = `
"<div data-v-52b2719a=\\"\\" class=\\"tooltip-container\\" data-testid=\\"tooltip\\">
  <div data-v-b9dd0015=\\"\\" data-v-52b2719a=\\"\\" class=\\"container info-icon\\"><svg data-v-b9dd0015-s=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"48\\" height=\\"48\\" viewBox=\\"0 96 960 960\\" class=\\"material-symbols-outlined\\">
      <path d=\\"M446.782 782.218h72.436V536h-72.436v246.218Zm33.25-311.739q17.642 0 29.544-11.638 11.903-11.638 11.903-28.841 0-18.689-11.92-30.584t-29.541-11.895q-18.257 0-29.877 11.895-11.62 11.895-11.62 30.301 0 17.557 11.935 29.159 11.934 11.603 29.576 11.603Zm.312 519.652q-86.203 0-161.506-32.395-75.302-32.395-131.741-88.833-56.438-56.439-88.833-131.738-32.395-75.299-32.395-161.587 0-86.288 32.395-161.665t88.745-131.345q56.349-55.968 131.69-88.616 75.34-32.648 161.676-32.648 86.335 0 161.779 32.604t131.37 88.497q55.926 55.893 88.549 131.452 32.623 75.559 32.623 161.877 0 86.281-32.648 161.575-32.648 75.293-88.616 131.478-55.968 56.186-131.426 88.765-75.459 32.579-161.662 32.579Z\\"></path>
    </svg></div>
  <div data-v-52b2719a=\\"\\" class=\\"tooltip\\" style=\\"display: none;\\">
    <div data-v-52b2719a=\\"\\" class=\\"tooltip-title text-small-semi-bold\\">Questo è un titolo</div>
    <div data-v-52b2719a=\\"\\" class=\\"tooltip-description text-small-regular\\">Questa è una descrizione</div>
  </div>
</div>"
`;

exports[`TooltipItem tests > should display tooltip with a title and a description when hovering the info icon 1`] = `
"<div data-v-52b2719a=\\"\\" class=\\"tooltip-container\\" data-testid=\\"tooltip\\">
  <div data-v-b9dd0015=\\"\\" data-v-52b2719a=\\"\\" class=\\"container info-icon\\"><svg data-v-b9dd0015-s=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"48\\" height=\\"48\\" viewBox=\\"0 96 960 960\\" class=\\"material-symbols-outlined\\">
      <path d=\\"M446.782 782.218h72.436V536h-72.436v246.218Zm33.25-311.739q17.642 0 29.544-11.638 11.903-11.638 11.903-28.841 0-18.689-11.92-30.584t-29.541-11.895q-18.257 0-29.877 11.895-11.62 11.895-11.62 30.301 0 17.557 11.935 29.159 11.934 11.603 29.576 11.603Zm.312 519.652q-86.203 0-161.506-32.395-75.302-32.395-131.741-88.833-56.438-56.439-88.833-131.738-32.395-75.299-32.395-161.587 0-86.288 32.395-161.665t88.745-131.345q56.349-55.968 131.69-88.616 75.34-32.648 161.676-32.648 86.335 0 161.779 32.604t131.37 88.497q55.926 55.893 88.549 131.452 32.623 75.559 32.623 161.877 0 86.281-32.648 161.575-32.648 75.293-88.616 131.478-55.968 56.186-131.426 88.765-75.459 32.579-161.662 32.579Z\\"></path>
    </svg></div>
  <div data-v-52b2719a=\\"\\" class=\\"tooltip\\" style=\\"display: none;\\">
    <div data-v-52b2719a=\\"\\" class=\\"tooltip-title text-small-semi-bold\\">Questo è un titolo</div>
    <div data-v-52b2719a=\\"\\" class=\\"tooltip-description text-small-regular\\">Questa è una descrizione</div>
  </div>
</div>"
`;

exports[`TooltipItem tests > should not display tooltip since the object there is no title nor description passed as props 1`] = `
"<div data-v-52b2719a=\\"\\" class=\\"tooltip-container\\" data-testid=\\"tooltip\\">
  <div data-v-b9dd0015=\\"\\" data-v-52b2719a=\\"\\" class=\\"container info-icon\\"><svg data-v-b9dd0015-s=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"48\\" height=\\"48\\" viewBox=\\"0 96 960 960\\" class=\\"material-symbols-outlined\\">
      <path d=\\"M446.782 782.218h72.436V536h-72.436v246.218Zm33.25-311.739q17.642 0 29.544-11.638 11.903-11.638 11.903-28.841 0-18.689-11.92-30.584t-29.541-11.895q-18.257 0-29.877 11.895-11.62 11.895-11.62 30.301 0 17.557 11.935 29.159 11.934 11.603 29.576 11.603Zm.312 519.652q-86.203 0-161.506-32.395-75.302-32.395-131.741-88.833-56.438-56.439-88.833-131.738-32.395-75.299-32.395-161.587 0-86.288 32.395-161.665t88.745-131.345q56.349-55.968 131.69-88.616 75.34-32.648 161.676-32.648 86.335 0 161.779 32.604t131.37 88.497q55.926 55.893 88.549 131.452 32.623 75.559 32.623 161.877 0 86.281-32.648 161.575-32.648 75.293-88.616 131.478-55.968 56.186-131.426 88.765-75.459 32.579-161.662 32.579Z\\"></path>
    </svg></div>
  <div data-v-52b2719a=\\"\\" class=\\"tooltip\\" style=\\"display: none;\\">
    <!--v-if-->
    <div data-v-52b2719a=\\"\\" class=\\"tooltip-description text-small-regular\\"></div>
  </div>
</div>"
`;
